url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FclImporter.java

```diff
@@ -0,0 +1,613 @@
+                ruleBlock.setEnabled(extractEnabled(line));
+            } else if ("RULE".equals(firstToken)) {
+                int ruleStart = line.indexOf(':');
+                if (ruleStart < 0) {
+                    ruleStart = "RULE".length();
+                }
+                String rule = line.substring(ruleStart + 1).trim();
+                ruleBlock.addRule(Rule.parse(rule, engine));
+            } else {
+                throw new RuntimeException(String.format(
+                        "[syntax error] keyword <%s> not recognized in line %s",
+                        firstToken, line));
+            }
+        }
+    }
+
+    protected TNorm extractTNorm(String line) {
+        List<String> token = Op.split(line, ":");
+        if (token.size() != 2) {
+            throw new RuntimeException("[syntax error] "
+                    + "expected property of type (key : value) in line: " + line);
+        }
+        String name = token.get(1).trim();
+        String className = null;
+        if ("MIN".equals(name)) {
+            className = Minimum.class.getSimpleName();
+        } else if ("PROD".equals(name)) {
+            className = AlgebraicProduct.class.getSimpleName();
+        } else if ("BDIF".equals(name)) {
+            className = BoundedDifference.class.getSimpleName();
+        } else if ("DPROD".equals(name)) {
+            className = DrasticProduct.class.getSimpleName();
+        } else if ("EPROD".equals(name)) {
+            className = EinsteinProduct.class.getSimpleName();
+        } else if ("HPROD".equals(name)) {
+            className = HamacherProduct.class.getSimpleName();
+        }
+        return FactoryManager.instance().tnorm().createInstance(className);
+    }
+
+    protected SNorm extractSNorm(String line) {
+        List<String> token = Op.split(line, ":");
+        if (token.size() != 2) {
+            throw new RuntimeException("[syntax error] "
+                    + "expected property of type (key : value) in line: " + line);
+        }
+        String name = token.get(1).trim();
+        String className = name;
+        if ("MAX".equals(name)) {
+            className = Maximum.class.getSimpleName();
+        } else if ("ASUM".equals(name)) {
+            className = AlgebraicSum.class.getSimpleName();
+        } else if ("BSUM".equals(name)) {
+            className = BoundedSum.class.getSimpleName();
+        } else if ("NSUM".equals(name)) {
+            className = NormalizedSum.class.getSimpleName();
+        } else if ("DSUM".equals(name)) {
+            className = DrasticSum.class.getSimpleName();
+        } else if ("ESUM".equals(name)) {
+            className = EinsteinSum.class.getSimpleName();
+        } else if ("HSUM".equals(name)) {
+            className = HamacherSum.class.getSimpleName();
+        }
+        return FactoryManager.instance().snorm().createInstance(className);
+    }
+
+    protected Term extractTerm(String line) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.